ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionProperties.java

```diff
@@ -301,40 +308,52 @@ void validate() throws IngestionClientException {
 
         String mappingReference = ingestionMapping.getIngestionMappingReference();
         IngestionMapping.IngestionMappingKind ingestionMappingKind = ingestionMapping.getIngestionMappingKind();
-        String message = null;
+        String message = "";
 
-        if (ingestionMapping.getColumnMappings() != null) {
-            if (StringUtils.isNotBlank(mappingReference)) {
-                message += String.format("Both mapping reference '%s' and column mappings were defined.", mappingReference);
-            }
-            for (ColumnMapping column : ingestionMapping.getColumnMappings()) {
-                if (!column.isValid(ingestionMappingKind)) {
-                    message += String.format("Column mapping '%s' is invalid", column.columnName);
-                }
-            }
-        } else if (StringUtils.isBlank(mappingReference)) {
+        if ((ingestionMapping.getColumnMappings() == null) && StringUtils.isBlank(mappingReference)) {
             if (dataFormat.isMappingRequired()) {
                 message += String.format("Mapping must be specified for '%s' format.", dataFormat.name());
```

Comment:
I don't think we should remove this until 3.0.1, just to be sure everything else is stable first. This would be a restriction relaxation, so that can be part of a patch version.

Reply:
This should be a comment in the code rather than on the PR

Reply:
I generally try to avoid TODOs in the code. Instead (I think we discussed this over Teams as well), I already made this change locally, waiting to create a PR for it until after this is merged, which we'll do concurrently with a discussion with Tzvia.